shopelectro/views/catalog.py

```diff
@@ -25,6 +26,62 @@ def get_products_count(request):
     return PRODUCTS_ON_PAGE_MOB if mobile_view else PRODUCTS_ON_PAGE_PC
 
 
+def get_view_type(request):
+    view_type = request.session.get('view_type', 'tile')
+    assert view_type in ['list', 'tile']
+    return view_type
+
+
+def get_catalog_context(request, category, raw_tags, page_number, per_page, sorting_index):
+    all_tags = newcontext.Tags(models.Tag.objects.all())
+
+    if raw_tags:
+        selected_tags = newcontext.tags.Checked404Tags(
+            newcontext.tags.ParsedTags(
+                tags=all_tags,
+                raw_tags=raw_tags,
+            )
+        )
+    else:
+        selected_tags = newcontext.Tags(models.Tag.objects.none())
+
+    products = newcontext.products.OrderedProducts(
```

Comment:
why here you are using contexts chaining throw nested calls, but below - throw Contexts args?

I mean this peace of code below
```
contexts = newcontext.Contexts(
	all_tags, paginated_products, images, brands, grouped_tags,
)
```

Reply:
Because these classes are decorators and they are semantically linked to the single entity, in this case it is `Product`
 
`Contexts` just aggregate different contexts together into a single context class